### ISSUE TITLE
500 error when trying to login / 'attempt to write a readonly database' in the logs

### ISSUE THREAD

**Issue by HarmlessSaucer** [babybuddy/babybuddy#411]
I recently started getting a 500 error when trying to access BabyBuddy.

At first I thought it was related to me using BabyBuddy behind NGINX Proxy Manager, so I updated to using `CSRF_TRUSTED_ORIGINS` in the docker-compose file, but the issue persisted.

```dockerfile
---
version: "2.1"
services:
  babybuddy:
    image: lscr.io/linuxserver/babybuddy
    container_name: babybuddy
    environment:
      - TZ=Europe/London
      - CSRF_TRUSTED_ORIGINS=https://babybuddy.mydomain.com
      - SECURE_PROXY_SSL_HEADER=True
      - DEBUG=1
    volumes:
      - /home/graham/docker/babybuddy:/config
    ports:
      - 8002:8000
    restart: unless-stopped
```

To test this, I turned off all the things related to SSL/NGINX Proxy Manager and accessed it directly through it's IP Address and got the same error.
I turned on debugging, and I noticed I was getting the following error:

```
attempt to write a readonly database

Request Method: | POST
-- | --
http://192.168.1.220:8002/login/
4.0.2
OperationalError
attempt to write a readonly database
```

I'm running this through Docker-Compose on a Ubuntu system, so I had a look at the permissions and couldn't see any issue there.  

Any ideas where I could start to look for an issue?  Everything was working fine yesterday and I didn't change anything which is very curious.

**Comment by cdubz** [comment]
Just to be sure, could you post the output `ls -lah /home/graham/docker/babybuddy/*`?

**Comment by HarmlessSaucer** [comment]
Of course!  Thanks :)

```
-rw-rw-r-- 1  911  911  308 Feb 17 21:38 /home/graham/docker/babybuddy/docker-compose.yml

/home/graham/docker/babybuddy/custom-cont-init.d:
total 8.0K
drwxr-xr-x 2 root root 4.0K Dec 24 00:03 .
drwxrwxr-x 6  911  911 4.0K Feb 17 21:38 ..

/home/graham/docker/babybuddy/custom-services.d:
total 8.0K
drwxr-xr-x 2 root root 4.0K Dec 24 00:03 .
drwxrwxr-x 6  911  911 4.0K Feb 17 21:38 ..

/home/graham/docker/babybuddy/data:
total 444K
drwxr-xr-x 2 911 911 4.0K Mar  5 14:03 .
drwxrwxr-x 6 911 911 4.0K Feb 17 21:38 ..
-rw-r--r-- 1 911 911 432K Mar  4 06:55 db.sqlite3

/home/graham/docker/babybuddy/media:
total 16K
drwxr-xr-x 4 911 911 4.0K Feb 17 10:08 .
drwxrwxr-x 6 911 911 4.0K Feb 17 21:38 ..
drwxr-xr-x 3 911 911 4.0K Feb 17 10:08 CACHE
drwxr-xr-x 3 911 911 4.0K Jan 29 00:12 child
```

**Comment by cdubz** [comment]
Just out of curiosity if you do `chmod 777 /home/graham/docker/babybuddy/data/db.sqlite3` does it work? Just trying to determine if its an actual permissions issue or something else that shows up as a permissions issue 🤔

**Comment by HarmlessSaucer** [comment]
Yea that was my next logical thought now I have a minute where TV is parenting!  Haha
```
-rwxrwxrwx 1 911 911 432K Mar  4 06:55 db.sqlite3
```

Still getting a 500 error when logging in.  I'm about to flip on debug mode again to see what happens.

**Comment by HarmlessSaucer** [comment]
Well it totally looks like that was it... Now I'm getting 'database or disk full', erorr but that's expected for this test volume I'm doing this on so not surprised.  

Not sure what happened to it - I just noticed the permissions on my whole babybuddy folder are wonky, as I can't edit the docker-compose I made now without sudo :thinking: Interesting.

Thank you for your patience and entertaining my silly issue! I definitely need to donate to you to support your efforts with this great project.
